keyboards/era/sirind/tomak/tomak.c

```diff
@@ -0,0 +1,219 @@
+// Copyright 2023 Hyojin Bak (@eerraa)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "tomak.h"
+#include "quantum.h"
+#include "eeprom.h"
+#include "transactions.h"
+
+tomak_config_t g_tomak_config;
+
+static void read_tomak_config_from_eeprom(tomak_config_t* config) {
+    config->raw = eeconfig_read_kb() & 0xffffffff;
+}
+
+static void write_tomak_config_to_eeprom(tomak_config_t* config) {
+    eeconfig_update_kb(config->raw);
+}
+
+uint8_t increment( uint8_t value, uint8_t step, uint8_t min, uint8_t max )
+{
+    int16_t new_value = value;
+    new_value += step;
+    return MIN( MAX( new_value, min ), max );
+}
+
+uint8_t decrement( uint8_t value, uint8_t step, uint8_t min, uint8_t max )
+{
+    int16_t new_value = value;
+    new_value -= step;
+    return MIN( MAX( new_value, min ), max );
+}
+
+void indicator_operation_toggle(void)
+{
+    g_tomak_config.indicator_toggle = g_tomak_config.indicator_toggle ? false : true;
+    write_tomak_config_to_eeprom(&g_tomak_config);
+}
+
+void indicator_override_toggle(void)
+{
+    g_tomak_config.indicator_override = g_tomak_config.indicator_override ? false : true;
+    write_tomak_config_to_eeprom(&g_tomak_config);
+}
+
+void indicator_brightness_increase(void)
+{
+    g_tomak_config.indicator_hsv.v = increment( g_tomak_config.indicator_hsv.v, 8, 0, 255 );
+    write_tomak_config_to_eeprom(&g_tomak_config);
+}
+
+void indicator_brightness_decrease(void)
+{
+    g_tomak_config.indicator_hsv.v = decrement( g_tomak_config.indicator_hsv.v, 8, 0, 255 );
+    write_tomak_config_to_eeprom(&g_tomak_config);
+}
+
+void indicator_hue_increase(void)
+{
+    g_tomak_config.indicator_hsv.h = increment( g_tomak_config.indicator_hsv.h, 8, 0, 255 );
+    write_tomak_config_to_eeprom(&g_tomak_config);
+}
+
+void indicator_hue_decrease(void)
+{
+    g_tomak_config.indicator_hsv.h = decrement( g_tomak_config.indicator_hsv.h, 8, 0, 255 );
+    write_tomak_config_to_eeprom(&g_tomak_config);
+}
+
+void indicator_sat_increase(void)
+{
+    g_tomak_config.indicator_hsv.s = increment( g_tomak_config.indicator_hsv.s, 8, 0, 255 );
+    write_tomak_config_to_eeprom(&g_tomak_config);
+}
+
+void indicator_sat_decrease(void)
+{
+    g_tomak_config.indicator_hsv.s = decrement( g_tomak_config.indicator_hsv.s, 8, 0, 255 );
+    write_tomak_config_to_eeprom(&g_tomak_config);
+}
+
```

Comment:
This all still looks to be VIA related code, or at least not very useful to just QMK.

Reply:
> This all still looks to be VIA related code, or at least not very useful to just QMK.

This is a code that allows you to control the indicator through a custom keycode regardless of VIA. T^T

May I know what is the acceptable range of using custom keycodes?
I am wondering if there is a way to be allowed to control the indicator through a custom keycode.